### ISSUE TITLE
Issue with leading semicolon in free-format

### ISSUE THREAD

**Issue by pjh40** [flang-compiler/f18#121]
Given an input of:
```
subroutine foo
  ; call bar
end subroutine
```
f18 is reporting
```
semi.f90:2:3: error: expected 'TYPE'
    ; call bar
    ^
semi.f90:2:3: in the context: TYPE statement
semi.f90:2:3: in the context: derived type definition
semi.f90:2:3: in the context: specification construct
semi.f90:2:3: in the context: declaration construct
semi.f90:2:3: in the context: specification part
semi.f90:1:1: in the context: SUBROUTINE subprogram
  subroutine foo
```
It would appear that the parsers are being fed an empty statement and they don't know what to do with it.  While the standard is clear that it is illegal to have a semicolon as the first non-blank character in an initial fixed format line [6.3.3.4-2], it is less clear for free form.  Given that repeated semicolons/spaces are equivalent to a single semicolon [6.3.2.5-2], I read that to mean that empty statements are ignored, and that rule should apply to the initial semicolon as well.  ifort and gfortran agree with this interpretation.  Regardless of wether you wish to support the empty initial statement or not, the error message is incorrect.

**Comment by klausler** [comment]
I can't find anything in the draft F'18 standard that would admit an empty statement (initial or not), but it seems like a safe and friendly extension.

The misleading error message stems from the error recovery mechanism retaining the messages from the alternative parse that went the furthest into the text; this is suboptimal when no parser made any progress.  I'm adding some default messages for these cases.

**Comment by klausler** [comment]
Both concerns are addressed with pull request #122.

**Comment by klausler** [comment]
Whenever it's convenient to do so, please confirm that this issue has been resolved to your satisfaction and close it, or let me know about any remaining problem.  Thanks for the bug report!

**Comment by pjh40** [comment]
With this patch, f18 passed my free source form torture test, including the leading semicolons.  Thanks for the quick fix!

**Comment by klausler** [comment]
Thanks for the confirmation.  Would it be possible for your test to become part of our regression suite?
